internal/host/validator.go

```diff
@@ -658,4 +659,157 @@ func (v *validator) printSufficientOrUnknownInstallationDiskSpeed(c *validationC
 	default:
 		return fmt.Sprintf("Unexpected status %s", status.String())
 	}
+
+}
+
+func (v *validator) hasSufficientNetworkLatencyRequirementForRole(c *validationContext) ValidationStatus {
+
+	if len(c.cluster.Hosts) == 1 {
+		// Single Node use case
+		return ValidationSuccess
+	}
+	if len(c.host.Connectivity) == 0 {
+		return ValidationPending
+	}
+	latency, err := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts)
+	if err != nil || len(latency) > 0 {
+		return ValidationFailure
+	}
+	return ValidationSuccess
+}
+
+func getHostnameAndRoleByIP(ip string, hosts []*models.Host) (string, models.HostRole, error) {
```

Comment:
Move to `host/common.go` as it could be useful in the future for somewhere else.

Reply:
Will do.